Single layer build instructions possibly broken

Some people are reporting that `yarn install` followed by `yarn dist` in the main repo alone (without the other 2 layers) fails with the error:

```
Module not found: Error: Can't resolve './component-index' in '/home/eiko/Coding/git/element-web/node_modules/matrix-react-sdk/src'
```

Since this is one of the setup paths we [describe in the docs](https://github.com/vector-im/element-web#building-from-source), we should either fix this or remove this path from the docs.

I was facing this issue today while building element-web from source, on Windows. Not sure if it matters, but I used local versions of matrix-js-sdk and matrix-react-sdk. Here is the log of this build:

```
PS C:\Users\Atom\Desktop\neo-matrix\element-web> yarn build
yarn run v1.22.5
$ yarn clean && yarn build:genfiles && yarn build:bundle
$ rimraf lib webapp
$ yarn reskindex && yarn build:res && yarn build:jitsi
$ reskindex -h src/header
Reskindex completed
$ node scripts/copy-res.js
$ node scripts/build-jitsi.js
Making webapp directory
Downloading Jitsi script
Done with Jitsi download
$ webpack --progress --bail --mode production
31% building 132/159 modules 27 active ...sktop\neo-matrix\matrix-react-sdk\src\settings\controllers\ThemeController.tsModuleNotFoundError: Module not found: Error: Can't resolve './component-index' in 'C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\src'
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\webpack\lib\Compilation.js:925:10
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\webpack\lib\NormalModuleFactory.js:401:22
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\webpack\lib\NormalModuleFactory.js:130:21
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\webpack\lib\NormalModuleFactory.js:224:22
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\neo-async\async.js:2830:7
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\neo-async\async.js:6877:13
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\webpack\lib\NormalModuleFactory.js:214:25
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\Resolver.js:213:14
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\UnsafeCachePlugin.js:44:7
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:27:1)
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\DescriptionFilePlugin.js:67:43
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:16:1)
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\RootPlugin.js:37:38
    at _next44 (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:8:1)
    at eval (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:31:1)
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:27:1)
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\DescriptionFilePlugin.js:67:43
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:16:1)
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\DirectoryExistsPlugin.js:27:15
    at C:\Users\Atom\Desktop\neo-matrix\element-web\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:85:15
resolve './component-index' in 'C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\src'
  using description file: C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\package.json (relative path: ./src)
    Field 'matrix_src_browser' doesn't contain a valid alias configuration
    using description file: C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\package.json (relative path: ./src/component-index)
      no extension
        Field 'matrix_src_browser' doesn't contain a valid alias configuration
        Field 'browser' doesn't contain a valid alias configuration
        C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\src\component-index doesn't exist
      .js
        Field 'matrix_src_browser' doesn't contain a valid alias configuration
        Field 'browser' doesn't contain a valid alias configuration
        C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\src\component-index.js doesn't exist
      .json
        Field 'matrix_src_browser' doesn't contain a valid alias configuration
        Field 'browser' doesn't contain a valid alias configuration
        C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\src\component-index.json doesn't exist
      .ts
        Field 'matrix_src_browser' doesn't contain a valid alias configuration
        Field 'browser' doesn't contain a valid alias configuration
        C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\src\component-index.ts doesn't exist
      .tsx
        Field 'matrix_src_browser' doesn't contain a valid alias configuration
        Field 'browser' doesn't contain a valid alias configuration
        C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\src\component-index.tsx doesn't exist
      as directory
        C:\Users\Atom\Desktop\neo-matrix\matrix-react-sdk\src\component-index doesn't exist
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

> The core issue here is that building from develop means you'll also have develop React SDK, but since you're attempting to operate at the top layer only, React SDK's component index won't exist.
> 
> One approach might be to also generates React SDK's component index as well from Element Web's indexing step.
> 
> For anyone facing this issue, you should be able to workaround it by running `(cd node_modules/matrix-react-sdk; yarn reskindex)` before trying to build.

Doing the above fixed the problem for me.

--- a/README.md
+++ b/README.md
@@ -298,6 +298,7 @@ cd element-web
 yarn link matrix-js-sdk
 yarn link matrix-react-sdk
 yarn install
+yarn reskindex
 yarn start
 ```
 

